src/node_metadata.cc

```diff
@@ -28,6 +28,9 @@
 #include <unicode/uversion.h>
 #endif  // NODE_HAVE_I18N_SUPPORT
 
+// version definitions of pure js deps
+#define UNDICI_VERSION "5.12.0"
```

Comment:
Instead of defining it here, it would probably be cleaner to simply create a `src/undici_version.h` and always regenerate the contents of that file in the update script, then include it here to get the version.

Reply:
Dont you think it would be beneficial to have all the definitions in a file for code readability purposes?

Reply:
I think it's easy enough to run `node -p process.versions` to see all the versions in one go. Also worth noting that currently the Undici version is not on this file, and nobody complained about readability.

Reply:
Fine enough, i am trying to update pr in accordance with all the inputs received

Reply:
Have changed as per the recommendations have moved the version definitions to separate files which are written freshly each time the update script runs